File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int size = arr.length < arr[0].length ? arr[0].length : arr.length;
        int[][] answer = new int[size][size];
        
        for (int i=0; i<arr.length; i++) {
            System.arraycopy(arr[i], 0, answer[i], 0, arr[i].length);
        }        
        
        return answer;
    }
}

File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    a = len(arr)
    b = len(arr[0])
    
    if a < b:
        for i in range(b-a):
            arr.append([0] * b)
    else:
        for i in range(a):
            arr[i] += [0] * (a-b) 
    
    return arr